### PR TITLE
Add auth scenes to the Router

### PR DESCRIPTION
**Pull request by VyacheslavDudin**
Also move interfaces into seperate files

### CODE REVIEW

**Comment by VyacheslavDudin** [comment]
Tested, user is creating in db

File: frontend/src/common/models/auth/ISignInFIelds.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+interface ISignInFields {
```
Comment (AlikRakhmonov) [inline frontend/src/common/models/auth/ISignInFIelds.ts:1]: name is unclear. make it  ILoginUser

File: frontend/src/common/models/auth/ISignUpFields.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+interface ISignUpFields {
```
Comment (AlikRakhmonov) [inline frontend/src/common/models/auth/ISignUpFields.ts:1]: https://github.com/BinaryStudioAcademy/bsa-2020-chatito/blob/dev/backend/src/common/models/user/IRegisterUser.ts

File: frontend/src/common/models/user/IUser.ts [hunk 3]
```diff
@@ -1,8 +1,10 @@
-export interface IUser {
+interface IUser {
```
Comment (AlikRakhmonov) [inline frontend/src/common/models/user/IUser.ts:1]: ?

File: frontend/src/common/models/user/IUserState.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import { IUser } from './IUser';
+
+interface IUserState {
+  data: IUser | null;
```
Comment (AlikRakhmonov) [inline frontend/src/common/models/user/IUserState.ts:4]:
```suggestion
  user?: IUser;
```

interface for state should be in reducer

File: frontend/src/components/InputField/InputField.tsx [hunk 5]
```diff
@@ -13,17 +13,23 @@ const InputField = ({ label, ...props }: IProps) => {
   const style = {
     width: '300px'
   };
+  const errorStyle = {
```
Comment (AlikRakhmonov) [inline frontend/src/components/InputField/InputField.tsx:16]: why don't you use scss?

File: frontend/src/common/models/user/IUser.ts (hunk 3 above)
Comment (AlikRakhmonov) [reply frontend/src/common/models/user/IUser.ts:1]: see chat

File: frontend/src/scenes/SignIn/index.tsx [hunk 6]
```diff
@@ -29,7 +29,7 @@ const SignIn: FunctionComponent<IProps> = ({ fetchUser }) => {
 
   return (
     <div className={styles.signIn}>
-      <h1 className="text-center p-5">Sign in</h1>
+      <h1 className={`text-center ${styles['signIn-header']}`}>Sign in</h1>
```
Comment (AlikRakhmonov) [inline frontend/src/scenes/SignIn/index.tsx:32]:
```suggestion
      <h1 className={`text-center ${styles.signInHeader]}`}>Sign in</h1>
```

File: frontend/src/services/authService.ts [hunk 7]
```diff
@@ -13,9 +15,15 @@ export const login = async (userInput: ISignInFields) => {
...
-    userData: JSON.stringify(userInput)
+    user: { ...userInput }
   };
   const response: ISignServerResponse & Response = await api.post('/api/auth/register', userData);
 
   return response.json();
 };
+
+export const fetchUser = async (): Promise<IUser> => {
+  const response: ISignServerResponse & Response = await api.get('/api/me/');
```
Comment (AlikRakhmonov) [inline frontend/src/services/authService.ts:26]:
```suggestion
  const response = await api.get<UseGenericHere>('/api/me');
```

File: frontend/src/sagas/user.ts [hunk 8]
```diff
@@ -1,4 +1,8 @@
 import { all, put, call, takeEvery } from 'redux-saga/effects';
+<<<<<<< HEAD
```
Comment (vlvereta) [inline frontend/src/sagas/user.ts:2]: wtf?
Resolve this and next lines please.

**Pull request merged**